### PR TITLE
[Pallas] Fix argmax/argmin with keepdim on 3D+ tensors

### PR DESCRIPTION
**Pull request by Aminsed**
## Summary
- Fix Pallas backend codegen crash when lowering `argmax`/`argmin` with `keepdim=True` on non-outermost dimensions of 3D+ tensors.
- Disambiguate partial-reduction axis selection for arg-reductions and keep behavior correct for 2D and 3D+ layouts.
- Add/keep coverage in `test/inductor/test_pallas.py` for the affected argmax/argmin keepdim cases.

## Context
This PR supersedes #174726, which became non-reopenable after a rebase/refresh attempt (GitHub returned "Could not open the pull request.").

## Test Plan
- `python test/inductor/test_pallas.py PallasCpuTests.test_argmax_argmin2_cpu_pallas`
- `python test/inductor/test_pallas.py PallasCpuTests.test_argmax_keepdim_3d`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: torch/_inductor/runtime/runtime_utils.py [hunk 1]
```diff
@@ -262,28 +268,69 @@ def pallas_partial_reduce(reduce_fn: Any, v: Any, pw_numel: int, red_numel: int)
     Args:
         reduce_fn: The reduction function to apply (e.g., jnp.sum, jnp.max)
         v: The input array to reduce
+<<<<<<< HEAD
```
Comment (oulgen) [inline torch/_inductor/runtime/runtime_utils.py]: please clean the merge markers

**Review by oulgen — COMMENTED** [review]
tagging @v0i0 since he was looking at something similar
